### PR TITLE
Allow `root` in `raw-options`

### LINKED ISSUE #16: setuptools-scm "root" parameter is ignored in raw-options
Consider this config in the pyproject.toml:

```toml
[tool.hatch.version]
raw-options = { root = ".." }
source = "vcs"
```

and such source tree:

```
.
├── .git
├── package1
|   ...
│   └── pyproject.toml
├── package2
|   ...
│   └── pyproject.toml
...
└── README.md
```

If I try to build a package with dynamic versioning then it will fail because it `setuptools-scm` is not able to find `.git` directory

```
LookupError: Error getting the version from source `vcs`: setuptools-scm was unable to detect version for /some/path/package1.

Make sure you're either building from a fully intact git repository or PyPI tarballs. Most other sources (such as GitHub's tarballs, a git checkout without the .git folder) don't contain the necessary metadata and will not work.
```

It happens due to this line https://github.com/ofek/hatch-vcs/blob/388c947133061a2056de9fe2f1102685a648f248/hatch_vcs/version_source.py#L56. I'm not sure what 's the intention of this but this prevents to work with projects where `pyproject.toml` is not located in the same directory with `.git`.

### PR DESCRIPTION
**Pull request by quarckster**
Fixes #16.

### CODE REVIEW

File: hatch_vcs/version_source.py [hunk 1]
```diff
@@ -46,14 +46,13 @@ def config_raw_options(self):
...
-    def get_version_data(self):
+    
+    @property
+    def setuptools_scm_config(self):
         from copy import deepcopy
 
-        from setuptools_scm import get_version
-
         config = deepcopy(self.config_raw_options)
-        config['root'] = self.root
```
Comment (RonnyPfannschmidt) [inline hatch_vcs/version_source.py:56]: I believe this fix is incomplete as root for hatch and setuptools_scm are distinct ideas

File: hatch_vcs/version_source.py [hunk 2]
```diff
@@ -46,14 +46,13 @@ def config_raw_options(self):
...
+        config['root'] = config.get('root', self.root)
```
Comment (ofek) [inline hatch_vcs/version_source.py:55]:
```suggestion
        config.setdefault('root', self.root)
```

**Review by ofek — CHANGES_REQUESTED** [review]

File: tests/test_version_config.py [hunk 3]
```diff
@@ -50,6 +50,11 @@ def test_not_table(self, new_project_basic):
         with pytest.raises(TypeError, match='option `raw-options` must be a table'):
             _ = version_source.config_raw_options
 
+    def test_root(self, new_project_basic):
+        config = {'raw-options': {'root': '..'}}
+        version_source = VCSVersionSource(new_project_basic, config)
+        assert version_source.setuptools_scm_config["root"] == ".."
```
Comment (ofek) [inline tests/test_version_config.py:56]:
```suggestion
        assert version_source.setuptools_scm_config['root'] == '..'
```

File: hatch_vcs/version_source.py [hunk 4]
```diff
@@ -46,14 +46,13 @@ def config_raw_options(self):
             self.__config_raw_options = raw_options
 
         return self.__config_raw_options
-
-    def get_version_data(self):
+
```
Comment (ofek) [inline hatch_vcs/version_source.py:49]:
```suggestion

```

File: hatch_vcs/version_source.py [hunk 5]
```diff
@@ -46,14 +46,12 @@ def config_raw_options(self):
             self.__config_raw_options = raw_options
 
         return self.__config_raw_options
-
-    def get_version_data(self):
+    @property
```
Comment (ofek) [inline hatch_vcs/version_source.py:49]:
```suggestion

    @property
```

**Pull request merged**